lib/screens/v2/disruption_diagram/model.ex

```diff
@@ -56,26 +61,179 @@ defmodule Screens.V2.DisruptionDiagram.Model do
           show_symbol: boolean()
         }
 
-  @type label :: ellipsis() | %{full: String.t(), abbrev: String.t()}
+  @type label :: label_map() | ellipsis()
+
+  @type label_map :: %{full: String.t(), abbrev: String.t()}
 
   # Literally the string "…", but you can't use string literals as types in elixir
   @type ellipsis :: String.t()
 
   # End labels have hardcoded presentation, so we just send an ID for the client to use in
   # a lookup.
   #
-  # TBD what these IDs will look like. We might just use parent station IDs.
+  # In most cases, the IDs are parent station IDs. For compound labels like
+  # "to Ashmont & Braintree", two IDs are joined with '+': "place-asmnl+place-brntn".
+  # For labels that don't use station names, we just use an agreed-upon string:
+  # "western_branches", "place-kencl+west", etc.
   #
   # The rest of the labels' presentations are computed based on the height of the end labels,
   # so we can send actual text for those--it will be dynamically resized to fit.
   @type end_label_id :: String.t()
 
   @type line_color :: :blue | :orange | :red | :green
 
+  @minimum_slot_count 6
+
+  @max_closure_count 8
+
+  # The max number of stops allowed in the gap when it needs to be collapsed
+  @collapsed_gap_max 2
+
   @doc "Produces a JSON-serializable map representing the disruption diagram."
-  # Update spec when this gets implemented!
-  @spec serialize(t()) :: nil
-  def serialize(_model) do
-    nil
+  @spec serialize(t()) :: {:ok, serialized_response()} | {:error, reason :: String.t()}
+  def serialize(localized_alert) when is_localized_alert(localized_alert) do
```

Comment:
Currently, it's not possible in the one place this is called.
I think it's ok for us to fail here if that happens, because such a case should be easy to catch in local development—I can't imagine a scenario where it would happen inconsistently, since we would have to conditionally pass a whole different map shape to this function for that to happen.